solana/solana-ibc/programs/solana-ibc/src/tests.rs

```diff
@@ -151,14 +163,85 @@ fn anchor_test_deliver() -> Result<()> {
         ibc::ClientMsg::CreateClient,
         ibc::MsgEnvelope::Client,
     );
+
+    let serialized_msg = message.try_to_vec().unwrap();
```

Comment:
With the chunks account being length-prefixed this would be:

```suggestion
    let serialized_msg = message.try_to_vec().unwrap();
    let serialized_msg = [
        &(serialized_msg.len() as u32).to_le_bytes()[..],
        serialized_msg.as_slice(),
    ].concat();
```